chrome-extension/content.js

```diff
@@ -120,23 +130,61 @@ function customCreateElement(type, classList, innerHTML = '') {
 const serverUrl = 'http://localhost:8080';
 // TODO: select between dev host and prod host
 
-async function addItem() {
+/**
+ * Returns a Promise that resolves to a <select> element,
+ * a multi-select dropdown menu of category options from fetching db
+ * @param {String} userId 
+ */
+async function getCategoryDropdown(userId) {
+  const categoryDropdown = customCreateElement('select', []);
+  categoryDropdown.setAttribute('multiple', true);
+
+  const defaultOption = customCreateElement('option', [], 'Option: Select a Category');
+  defaultOption.value = '';
+  defaultOption.disabled = true;
+  defaultOption.selected = true; 
+  categoryDropdown.options.add(defaultOption);
+
+  try {
+    const response = await fetch(serverUrl + '/category/getCategories/' + userId);
+    const categories = await response.json();
+    categories.map((category) => {
+      const newOption = customCreateElement('option', [], category.title);
+      newOption.value = category._id;
+      categoryDropdown.options.add(newOption);
+    });
+  } catch (error) {
+    // TODO: Handle this error? 
+  }
+  return categoryDropdown
+}
+
+/**
+ * Add new item to databse
+ */
+async function addItem() {  
   newItem = {
-    sourceLink: 'www.googe.com', // TODO: get actual sourceLink
+    sourceLink: window.location.toString(),
     placesId: 'something', // TODO: get actual placesId
-    userId: '5f050952f516f3570ee26724', // TODO: get actual userID
+    userId: '66616b652d757365722d6964', // TODO: get actual userID
```

Comment:
While you're using a dummy userId, perhaps you can use a constant global variable and just make one comment there to get the actual userId

Reply:
done